OWML.ModHelper.Input/ModInputHandler.cs

```diff
@@ -0,0 +1,440 @@
+                var toRemove = new List<IModInputCombination>();
+                foreach (IModInputCombination combo in _singlePressed)
+                    if (!IsPressed(combo))
+                    {
+                        ((ModInputCombination)combo).SetPressed(false);
+                        toRemove.Add(combo);
+                    }
+                foreach (IModInputCombination combo in toRemove)
+                    _singlePressed.Remove(combo);
+            }
+            List<KeyCode> keys = ((ModInputCombination)combination).GetSingles();
+            foreach (KeyCode key in keys)
+            {
+                if (Input.GetKey(key) && (!ShouldIgnore(key) || _singlePressed.Contains(combination)))
+                {
+                    _singlePressed.Add(combination);
+                    _timeout[(int)key] = Time.realtimeSinceStartup;
+                    ((ModInputCombination)combination).SetPressed();
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private bool IsNewlyPressed_Single(IModInputCombination combination, bool keep = false)
+        {
+            return IsPressed_Single(combination) && combination.IsFirst(keep);
+        }
+
+        private bool WasTapped_Single(IModInputCombination combination)
+        {
+            return (!IsPressed_Single(combination)) && (Time.realtimeSinceStartup - combination.GetLastPressedMoment() < _tapKeep) && (combination.GetPressDuration() < _tapDuration);
+        }
+
+        private bool IsNewlyReleased_Single(IModInputCombination combination, bool keep = false)
+        {
+            return (!IsPressed_Single(combination)) && combination.IsFirst(keep) && (Time.realtimeSinceStartup - combination.GetLastPressedMoment() < _tapKeep);
+        }
+
+        public bool IsPressed(IModInputCombination combination)
+        {
+            return IsPressed_Combo(combination) || IsPressed_Single(combination);
+        }
+
+        public bool IsNewlyPressed(IModInputCombination combination, bool keep = false)
+        {
+            return IsPressed(combination) && combination.IsFirst(keep);
+        }
+
+        public bool WasTapped(IModInputCombination combination)
+        {
+            return (!(IsPressed_Combo(combination) || IsPressed_Single(combination))) && (Time.realtimeSinceStartup - combination.GetLastPressedMoment() < _tapKeep) && (combination.GetPressDuration() < _tapDuration);
+        }
+
+        public bool WasNewlyReleased(IModInputCombination combination, bool keep = false)
+        {
+            return (!(IsPressed_Combo(combination) || IsPressed_Single(combination))) && (Time.realtimeSinceStartup - combination.GetLastPressedMoment() < _tapKeep);
+        }
+
+        private Int64 ParseCombination(string combo, bool forRemoval = false)
+        {
+            combo = combo.Trim().Replace("ctrl", "control");
+            int[] curcom = new int[7];
```

Comment:
```suggestion
            int[] currentCombination = new int[7];
```

Reply:
Also why 7? Again the same 7?

Reply:
done, although renamed to thisCombination instead